corehq/apps/cleanup/management/commands/fix_forms_with_missing_xmlns.py

```diff
@@ -0,0 +1,175 @@
+                    f.write("Set new xmlns on submission {}\n".format(xform_instance._id))
+                else:
+                    app = Application.get(xform_instance.app_id)
+                    forms_without_xmlns = get_forms_without_xmlns(app)
+                    if len(forms_without_xmlns) == 1:
+                        form = forms_without_xmlns[0]
+                        if not xforms_with_real_xmlns_possibly_exist(app._id, form):
+                            num_fixed += 1
+                            new_xmlns = generate_random_xmlns()
+                            new_xmlnss[xform_instance.app_id] = new_xmlns
+                            f.write("New xmlns for form {form.unique_id} in app {app._id} is {new_xmlns}\n".format(
+                                form=form,
+                                app=app,
+                                new_xmlns=new_xmlns
+                            ))
+                            set_xmlns_on_form(form, new_xmlns, app, dry_run)
+                            set_xmlns_on_submission(xform_instance, new_xmlnss[xform_instance.app_id], dry_run)
+                            f.write("Set new xmlns on submission {}\n".format(xform_instance._id))
+
+
+def get_submissions_without_xmlns():
+    # TODO: This view won't get archived forms etc
+    submissions = XFormInstance.get_db().view(
+        'couchforms/by_xmlns',
+        key="undefined",
+        include_docs=False,
+        reduce=False,
+    ).all()
+    total_submissions = len(submissions)
+    submission_id_generator = (s['id'] for s in submissions)
+    doc_generator = (XFormInstance.wrap(i) for i in iter_docs(XFormInstance.get_db(), submission_id_generator))
+    return total_submissions, doc_generator
+
+
+def xforms_with_real_xmlns_possibly_exist(app_id, form):
+    """
+    Return True if there exist xforms submitted to the given app against a form
+    with the same name as the given name that have a real xmlns.
+
+    If this is True, it could indicate that the form had an xlmns, had submissions,
+    then lost the xmlns. If that's the case, we would want to give the submissions
+    this xmlns, not a new random one.
+    """
+    for form_name in form.name.values():
+        query = (FormES()
+                 .term('form.@name', form_name)
+                 .app(app_id)
+                 .filter(NOT(xmlns('undefined'))))
+        if query.count():
+            return True
+    return False
+
+
+def set_xmlns_on_submission(xform_instance, xmlns, dry_run):
+    """
+    Set the xmlns on an XFormInstance, and the save the document.
+    """
+    old_xml = xform_instance.get_xml()
+    assert old_xml.count('xmlns="undefined"') == 1
+    new_xml = old_xml.replace('xmlns="undefined"', 'xmlns="{}"'.format(xmlns))
+    if not dry_run:
+        replace_xml(xform_instance, new_xml)
+
+    xform_instance.xmlns = xmlns
+    xform_instance.form['@xmlns'] = xmlns
+    if not dry_run:
+        xform_instance.save()
```

Comment:
Do you have an order-of-magnitude estimate of how many forms are affected?  If it's massive, it could be worth looking into saving in chunks.